lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -139,9 +138,7 @@ public function get(QueryCacheKey $key, ResultSetMapping $rsm, array $hints = []
                     continue;
                 }
 
-                $generateKeys = static function ($id) use ($assocMetadata): EntityCacheKey {
-                    return new EntityCacheKey($assocMetadata->rootEntityName, $id);
-                };
+                $generateKeys = static fn ($id): EntityCacheKey => new EntityCacheKey($assocMetadata->rootEntityName, $id);
```

Comment:
The second argument of `EntityCacheKey` is an array.

```suggestion
                $generateKeys = static fn (array $id): EntityCacheKey => new EntityCacheKey($assocMetadata->rootEntityName, $id);
```